src/unix/posix.h

```diff
@@ -80,4 +80,14 @@ GIT_INLINE(int) p_futimes(int f, const struct p_timeval t[2])
 # define p_futimes futimes
 #endif
 
+#ifdef HAVE_REGCOMP_L
+#include <xlocale.h>
+GIT_INLINE(int) p_regcomp(regex_t *preg, const char *pattern, int cflags)
+{
+	return regcomp_l(preg, pattern, cflags, (locale_t) 0);
```

Comment:
Unfortunately, I can't find anything either.

Initially, I created a new locale through `newlocale(LC_ALL_MASK, "C", (locale_t) 0);`, but I tested using `(locale_t) 0` directly and that seems to work as well.

We could change it back to using `newlocale` and storing that locale somewhere if you think that'd be safer?